rfcs/01780_policy_delegation.md

```diff
@@ -0,0 +1,308 @@
+JSON file with the schema described below.
+
+For instance, Google may publish a delegation manifest under
+`https://google.com/oak/manifests/count_people_from_pixels.json` with the
+following contents:
+
+```json
+{
+  "trusted": [
+    {
+      "type": "manifest",
+      "value": "https://eff.com/oak/manifests/count_people_from_pixels.json"
+    },
+    {
+      "type": "wasm_module",
+      "value": "65c296d60c077db03c7ec90c60107a52221c14bb"
+    },
+    {
+      "type": "wasm_module",
+      "value": "cf26b96746311908f8fe40340a8b6c1f06f5a1b5"
+    }
+  ]
+}
+```
+
+- ~~The `description` field contains a human-readable description of the entries
+  that the manifest refers to.~~ This was considered but then removed, so that
+  it may not be misused by showing it to users in the wrong context.
+- The `trusted` field contains a list of object that are trusted by the
+  publisher to conform to the abstract policy identified by the manifest. Each
+  object in the `trusted` list is composed of the following fields:
+  - `type`: the category of the object (each list of trusted object may have
+    objects of various types); some example values of the `type` fields are:
+    - `manifest`
+    - `wasm_module`
+    - `intel_sgx_mrenclave`
+  - `value`: a string representing a value that uniquely identifies the trusted
+    object; commonly this would be a hash of the object, hex or base64 encoded.
+
+It is possible to include / chain manifests by including the URL of a target
+manifest as a trusted object itself, in which case the chain of URLs is
+recursively expanded at runtime by Oak.
+
+Note that the manifest does not make any _formal_ guarantees about the inclusion
+policy for the objects it refers to; **the meaning of the URL that identifies
+the manifest must still be agreed upon informally and out-of-band, and does not
+have any inherent meaning to Oak**. A manifest URL may or may not be human
+readable at all; in any case it is **not** expected to be displayed to users
+directly. ~~The `description` field of the manifest may be used in the context
+of displaying to the user, but the link between that and any actual property of
+the trusted objects is completely up to the publisher, and based on trusting the
+publisher alone.~~
+
+As an example, technically there is nothing stopping Google from (perhaps at
+some point in the future) repurposing the manifest identified by
+`https://google.com/oak/manifests/count_people_from_pixels.json` to publish
+hashes of nodes that count the number of cats instead of the number of people.
+The only understanding of what the list of hashes published in a manifest "mean"
+is outside of the manifest system itself. It may include formal or informal
+guarantees about functionality or policies that the objects are subject to, but
+those need to be published separately.
+
+For instance, Google may have a separate web page in which expectations of what
```

Comment:
How will users decide which URLs they trust?

Reply:
They would browse some website (e.g. something that google.com creates for this purpose) and browse around until they find something that they trust. I expect these will be developers, not end users, for now. It is not much different from browsing GitHub to decide where to download a piece of source code, or browsing crates.io to decide what crate to depend on, but it's more decentralized in that it depends only on URLs.